join-reorder.md

```diff
@@ -43,13 +46,26 @@ In this case only three tables are joined, so TiDB gets the final join result.
 
 ![join-reorder-3](/media/join-reorder-3.png)
 
-The above process is the Join Reorder algorithm currently used in TiDB.
+## Example: the dynamic programming algorithm of Join Reorder
+
+Taking the preceding three tables (t1, t2, and t3) as an example again, the dynamic programming algorithm can enumerate all possibilities. Therefore, comparing with the greedy algorithm, which must start with the `t1` table (the table with the least rows), the dynamic programming algorithm can enumerate a join order as follows:
+
+! [join-reorder-4](/media/join-reorder-4.png)
+
+When this choice is better than the greedy algorithm, the dynamic programming algorithm can choose a better join order.
+
+Because all possibilities are enumerated, the dynamic programming algorithm consumes more time and is more susceptible to statistics.
+
+## Control the Join Reorder algorithms
+
+The Join Reorder algorithms are controlled by the [`tidb_opt_join_reorder_threshold`](/system-variables.md#tidb_opt_join_reorder_threshold) variable. If the number of nodes participating in Join Reorder is greater than this threshold, TiDB uses the greedy algorithm. Otherwise, TiDB uses the dynamic programming algorithm.
```

Comment:
```suggestion
The selection of the TiDB Join Reorder algorithm is controlled by the [`tidb_opt_join_reorder_threshold`](/system-variables.md#tidb_opt_join_reorder_threshold) variable. If the number of nodes participating in Join Reorder is greater than this threshold, TiDB uses the greedy algorithm. Otherwise, TiDB uses the dynamic programming algorithm.
```